xbmc/video/videosync/VideoSyncCocoa.cpp

```diff
@@ -33,6 +33,28 @@
 #if defined(TARGET_DARWIN_OSX)
 #include <QuartzCore/CVDisplayLink.h>
 #include "osx/CocoaInterface.h"
+#endif
+
+void CVideoSyncCocoa::OnLostDevice()
+{
+  if (!m_displayLost)
+  {
+    m_displayLost = true;
+    m_lostEvent.Wait();
```

Comment:
Currently I don't have a Mac so I can only guess.
The idea is that after a change of refresh rate windowing fires the OnResetDevice event which makes Run() terminate and VideoRefClock will reset and init again. As long as the implementation like DRM can run independently from windowing OnLostDevice is not needed.